src/lib/optimizer/strategy/predicate_split_up_rule.cpp

```diff
@@ -1,12 +1,62 @@
 #include "predicate_split_up_rule.hpp"
 
+#include "expression/binary_predicate_expression.hpp"
 #include "expression/expression_utils.hpp"
 #include "expression/logical_expression.hpp"
+#include "expression/value_expression.hpp"
 #include "logical_query_plan/lqp_utils.hpp"
 #include "logical_query_plan/union_node.hpp"
 
 namespace opossum {
 
+namespace {
+bool can_use_union_all(const std::vector<std::shared_ptr<AbstractExpression>>& predicates) {
+  // Optimization: The ExpressionReductionRule transforms `x NOT LIKE 'foo%'` into `x < 'foo' OR x >= 'fop'`. For this
+  // special case, we know that the two OR arguments are mutually exclusive. In this case, we do not need to use
+  // SetOperationMode::Positions but can use SetOperationMode::All. This surely applies to other cases, too, but those
+  // are not covered yet. Note that this is not equal to XOR. Here, we know that both sides cannot be true at the same
+  // time. XOR would require us to check if both are true and discard the row if they are.
+  if (predicates.size() != 2) return false;
+
+  const auto first_binary_predicate = std::dynamic_pointer_cast<BinaryPredicateExpression>(predicates[0]);
+  const auto second_binary_predicate = std::dynamic_pointer_cast<BinaryPredicateExpression>(predicates[1]);
+  if (!first_binary_predicate || !second_binary_predicate) return false;
+
+  // Check for pattern `col_x < 'val' AND col_x >= 'vam'`
+  if (first_binary_predicate->predicate_condition != PredicateCondition::LessThan ||
+      second_binary_predicate->predicate_condition != PredicateCondition::GreaterThanEquals) {
+    // Wrong predicates
+    return false;
+  }
```

Comment:
Maybe check the same after swapping predicates?

Reply:
Nah, that would make this optimization, which is only used for `NOT LIKE`, too complicated. If it ever comes up somewhere else, we can still check how to expand this rule.